gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -160,7 +172,16 @@ private void showAddFeatureDialog(Pair<ImmutableList<Layer>, Point> args) {
     ImmutableList<Layer> layers = args.first;
     Point point = args.second;
     addFeatureDialogFragment.show(
-        layers, getChildFragmentManager(), layer -> viewModel.addFeature(layer, point));
+        layers,
+        getChildFragmentManager(),
+        layer -> {
+          if (layer.getContributorsCanAdd().contains(FeatureType.POINT)
+              && layer.getContributorsCanAdd().contains(FeatureType.POLYGON)) {
+            showFeatureTypeDialog(layer, point);
+          } else {
```

Comment:
This assumes if contributors can add only one feature type, it will be a point. What if it's only a polygon?

Reply:
updated added check for point and polygon seperately.